MH-13208, Create a short contributor guide

lkiesow opened the pull request:
This patrch tries to create a short contributor guide, linking the
development process documentation but avoiding its lengty explanations.
The guide is in Github's default contributing guideline formnat and
should hence be pointed out by Github:

  https://help.github.com/articles/setting-guidelines-for-repository-contributors/

The guide intentionally tries to avoid mentioning any special cases and
points this out to readers. If necessary, they can still be found in the
developer documentation but most of the time, just sticking to this
guide should be fine.

The guide is meant especially for new, maybe even one-time contributors
which likely would not have read the long documentation before. Using
this, they should be able to quickly get an overview of how to create
proper pull requests for Opencast.

lkiesow committed:
MH-13208, Create a short contributor guide

This patrch tries to create a short contributor guide, linking the
development process documentation but avoiding its lengty explanations.
The guide is in Github's default contributing guideline formnat and
should hence be pointed out by Github:

  https://help.github.com/articles/setting-guidelines-for-repository-contributors/

The guide intentionally tries to avoid mentioning any special cases and
points this out to readers. If necessary, they can still be found in the
developer documentation but most of the time, just sticking to this
guide should be fine.

The guide is meant especially for new, maybe even one-time contributors
which likely would not have read the long documentation before. Using
this, they should be able to quickly get an overview of how to create
proper pull requests for Opencast.

staubesv commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+-------------
+
+Opencast uses [Jira](https://opencast.jira.com) for tracking issues. Each pull request should be accompanied by a ticket
+in Jira. The issue identifier should also be used in the title of the pull request and the commits. E.g.: `MH-12345,
```
@lkiesow The comma after the issue number seems somewhat superfluous. What do you think about "MH-12345 Description" instead of "MH-12345, Description"?

lkiesow replied:
This is the exact example from the development process docs. Also, to be honest, I like to have a separator between the ticket identifier and the summary. These are two different parts after all. But then, this is also just an example. If you like me to, I can remove it though, to just leave the choice to the contributor.

staubesv replied:
No strong opinion from my side, just wanted to mention it.

lkiesow replied:
I've simply removed the suggestion. There should be enough real-world examples to follow.

gregorydlogan commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+passing.
+
+Additionally, a reviewer will be assigned to your pull request to ensure that there are no further issues. Once
+everything is fine, the reviewer will merge the pull request.
```
"It may take some time before a review is assigned.  This is normal, so do not be concerned." ?

lkiesow replied:
Added

gregorydlogan commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+---------
+
+- [ ] Jira ticket created
+- [ ] Pull request has a proper title and description
```
Can we add a section about tagging your commits with the JIRA ticket as well?  That makes life so much easier.

lkiesow replied:
Done

gregorydlogan reviewed (approved):
Minor nitpicks, but looks good to me.

lkiesow committed:
MH-13208, Minor adjustments to CONTRIBUTING.md

- Remove PR title suggestion
- Add note about delayed reviews
- Stress usage of Jira ticket identifiers
